### PR TITLE
Test pypi catalyst

### PR DESCRIPTION
**Pull request by albi3ro**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/workflows/install_deps/action.yml [hunk 1]
```diff
@@ -36,6 +36,11 @@ inputs:
     description: Indicate if PennyLane-Lightning should be installed from the master branch
     required: false
     default: 'true'
+  install_catalyst_test_pypi:
+    description: Indicate if catalyst should be installed from the test pypi location
+    required: false
+    type: boolean
+    default: false
```
Comment (albi3ro) [inline .github/workflows/install_deps/action.yml:43]:
```suggestion
    default: 'false'
```

File: .github/workflows/interface-unit-tests.yml [hunk 2]
```diff
@@ -335,6 +335,38 @@ jobs:
...
+    with:
+      job_name: catalyst-tests (${{ matrix.python-version }})
+      branch: ${{ inputs.branch }}
+      coverage_artifact_name: catalyst-tests-coverage
+      python_version: ${{ matrix.python-version }}
+      pipeline_mode: ${{ inputs.pipeline_mode }}
+      install_jax: true
+      install_tensorflow: false
+      install_pytorch: false
+      install_pennylane_lightning_master: True
```
Comment (albi3ro) [inline .github/workflows/interface-unit-tests.yml:364]:
```suggestion
      install_pennylane_lightning_master: true
      install_catalyst_test_pypi: true
```

File: .github/workflows/install_deps/action.yml [hunk 3]
```diff
@@ -100,6 +105,11 @@ runs:
       if: inputs.install_pennylane_lightning_master == 'true'
       run: pip install -i https://test.pypi.org/simple/ PennyLane-Lightning --pre --upgrade
 
+    - name: Install Catalyst from Test Pypi
+      shell: bash
+      if: inputs.install_catalyst_test_pypi
```
Comment (rashidnhm) [inline .github/workflows/install_deps/action.yml:110]:
```suggestion
      if: inputs.install_catalyst_test_pypi == 'true'
```

Comment (albi3ro) [reply .github/workflows/install_deps/action.yml:110]: thanks!

File: .github/workflows/install_deps/action.yml [hunk 4]
```diff
@@ -36,6 +36,11 @@ inputs:
     description: Indicate if PennyLane-Lightning should be installed from the master branch
     required: false
     default: 'true'
+  install_catalyst_test_pypi:
+    description: Indicate if catalyst should be installed from the test pypi location
+    required: false
+    type: boolean
```
Comment (AmintorDusko) [inline .github/workflows/install_deps/action.yml:42]:
```suggestion
```

**Comment by trbromley** [comment]
@albi3ro @rashidnhm shall we get this reviewed and merged? There's no Shortcut story associated so it's hard to know where this sits in our Q3 plan.

**Comment by Alex-Preciado** [comment]
@albi3ro do we still need this?

**Pull request closed without merging**